### PR TITLE
Homework07 jdi

### PR DESCRIPTION
**Pull request by IuriiRadomskii**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw7/tests/MetalsAndColorsTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+public class MetalsAndColorsTest implements TestInit {
+
+    @DataProvider
+    public Object[][] getTestData() {
+        return JsonParser.getTestData();
+    }
+
+    @Test
+    public void loginTest() {
+        homePage.open();
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/tests/MetalsAndColorsTest.java:27]: mb need to use this methods inside login() ?

File: src/test/java/com/epam/tc/hw7/site/pages/BasePage.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.epam.tc.hw7.site.pages;
+
+import com.epam.jdi.light.elements.composite.WebPage;
+import com.epam.tc.hw7.enums.HeaderNavigationButtons;
+import com.epam.tc.hw7.site.components.Header;
+
+public class BasePage extends WebPage {
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/site/pages/BasePage.java:7]: what do this page ?

File: src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorsMainSection.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.Css;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.JDropdown;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.UI;
+import com.epam.jdi.light.ui.html.elements.common.Button;
+import com.epam.tc.hw7.data.Info;
+import java.util.List;
+import org.openqa.selenium.WebElement;
+
+@Css(".form")
+public class MetalAndColorsMainSection extends Section {
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorsMainSection.java:15]: mb it's Form ?

**Review by cold-flame — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw7/site/pages/BasePage.java (hunk 2 above)
Comment (IuriiRadomskii) [reply src/test/java/com/epam/tc/hw7/site/pages/BasePage.java:7]: This class contains common fields and methods for all pages. For example header, left-side navigation menu etc. This class implements "DRY" principle.

File: src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorsMainSection.java (hunk 3 above)
Comment (IuriiRadomskii) [reply src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorsMainSection.java:16]: That's right. I refactored it. And override in-built Form<T> method fill().

File: src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorForm.java [hunk 4]
```diff
@@ -12,10 +13,11 @@
 import org.openqa.selenium.WebElement;
 
 @Css(".form")
-public class MetalAndColorsMainSection extends Section {
+public class MetalAndColorForm extends Form<Info> {
 
     public Summary summary;
 
+    //@Css(".vertical-group .checkbox") public List<WebElement> elements;
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorForm.java:20]: Remove comment

File: src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorForm.java [hunk 5]
```diff
@@ -64,6 +67,10 @@ private void selectElements(List<String> elementsToClick) {
         }
     }
 
+    private void selectElements() {
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorForm.java:70]: Why this method is empty ?

File: src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/Summary.java [hunk 6]
```diff
@@ -18,6 +18,9 @@
     @FindBy(id = "calculate-button")
     private Button calculateBtn;
 
+    public Summary() {
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/Summary.java:21]: why this method is empty ?

**Review by cold-flame — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw7/data/Info.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+    private static final List<String> summaryData = List.of("4", "3");
+    private static final List<String> elementsData = List.of("Water", "Fire");
+    private static final String colorData = "Red";
+    private static final String metalsData = "Gold";
+    private static final List<String> vegetablesData = List.of("Cucumber", "Onion");
+    public static final Info TEST_DATA = new Info(summaryData,
+        elementsData, colorData, metalsData, vegetablesData);
+
+    public List<String> summary;
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/data/Info.java:17]: Why you use access modifiers public for this fields? You also create methods (getters) with public access modifiers, e.g. getSummary(). Change public -> private for fields

**Review by cold-flame — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorForm.java (hunk 4 above)
Comment (IuriiRadomskii) [reply src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorForm.java]: removed

File: src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorForm.java (hunk 5 above)
Comment (IuriiRadomskii) [reply src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/MetalAndColorForm.java]: removed

File: src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/Summary.java (hunk 6 above)
Comment (IuriiRadomskii) [reply src/test/java/com/epam/tc/hw7/site/components/pages/metalsandcolors/Summary.java:21]: also removed

File: src/test/java/com/epam/tc/hw7/data/Info.java (hunk 7 above)
Comment (IuriiRadomskii) [reply src/test/java/com/epam/tc/hw7/data/Info.java]: done, private fields was made public

**Review by cold-flame — APPROVED** [review]

**Pull request merged**